appverifier/src/main/java/com/android/mdl/appreader/MainActivity.kt

```diff
@@ -46,6 +57,31 @@ class MainActivity : AppCompatActivity() {
             addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
         }
         mPendingIntent = PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_IMMUTABLE)
+
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+        if (ActivityCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            requestPermissions(
+                listOf(
+                    Manifest.permission.ACCESS_FINE_LOCATION
+                ).toTypedArray(),
+                LOCATION_REQUEST_CODE
+            )
+            return
+        }
+
+        fusedLocationClient.lastLocation.addOnSuccessListener { lastKnownLocation ->
+            lastKnownLocation?.let{
```

Comment:
if we obtain a non-null last known Location then record the lat/long in the presentation log store so we can attach them to the log entry as it's about to be persisted